src/coreclr/nativeaot/System.Private.CoreLib/src/System/Diagnostics/StackTrace.CoreRT.cs

```diff
@@ -26,16 +27,32 @@ internal StackTrace(IntPtr[] ipAddresses, int startIndex, int endIndex, bool nee
         /// <summary>
         /// Initialize the stack trace based on current thread and given initial frame index.
         /// </summary>
+        [MethodImplAttribute(MethodImplOptions.NoInlining)]
         private void InitializeForCurrentThread(int skipFrames, bool needFileInfo)
         {
+            const int SystemDiagnosticsStackDepth = 2;
+
             int frameCount = -RuntimeImports.RhGetCurrentThreadStackTrace(Array.Empty<IntPtr>());
             Debug.Assert(frameCount >= 0);
             IntPtr[] stackTrace = new IntPtr[frameCount];
             int trueFrameCount = RuntimeImports.RhGetCurrentThreadStackTrace(stackTrace);
             Debug.Assert(trueFrameCount == frameCount);
-            InitializeForIpAddressArray(stackTrace, skipFrames, frameCount, needFileInfo);
+            if (trueFrameCount > 0 && IsRhGetCurrentThreadStackTraceFrame(stackTrace[0]))
+            {
+                skipFrames++;
```

Comment:
Can we do accurate accounting for `skipFrames` so that `IsRhGetCurrentThreadStackTraceFrame` is not necessary at all?

Reply:
Just read your other comment about this.